bin/rucio

```diff
@@ -2029,6 +2029,61 @@ def delete_metadata(args):
     raise NotImplementedError
 
 
+@exception_handler
+def get_did_meta(args):
+    """
+    %(prog)s get-generic-metadata [options] <field1=value1 field2=value2 ...>
+
+    Get data identifier metadata
+    """
+    client = get_client(args)
+    for did in args.dids:
+        scope, name = extract_scope(did)
+        meta = client.get_did_meta(scope=scope, name=name)
+        for k in meta:
+            print('%s: %s' % (k, meta[k]))
+    return SUCCESS
+
+
+@exception_handler
+def add_did_meta(args):
+    """
+    %(prog)s add-did-meta [options] <field1=value1 field2=value2 ...>
+
+    Add or update data identifier metadata
+    """
+    client = get_client(args)
+    scope, name = extract_scope(args.did)
+    client.add_did_meta(scope=scope, name=name, key=args.key, value=args.value)
+    return SUCCESS
+
+
+@exception_handler
+def delete_did_meta(args):
+    """
+    %(prog)s delete-did-meta [options] <field1=value1 field2=value2 ...>
+
+    Add or update data identifier metadata
+    """
+    client = get_client(args)
+    scope, name = extract_scope(args.did)
+    client.delete_did_meta(scope=scope, name=name, key=args.key)
+    return SUCCESS
+
+
+@exception_handler
+def list_dids_by_meta(args):
+    """
+    %(prog)s list-dids-by-meta [options] <field1=value1 field2=value2 ...>
+
+    List dids by metadata key value
+    """
+    client = get_client(args)
+    select = {args.key: args.value}
```

Comment:
maybe i misinterpret something, but this will only work a single key value pair?

user should be able to do something like this:

`rucio list-dids-by-meta key1=value1 key2=value2 key3=value`